src/cmd/add_server.go

```diff
@@ -184,14 +184,12 @@ func (c addServerCmdConfig) Run() {
 
 		clientPeerConfigE2EE, err := clientConfigE2EE.AsPeer()
 		check("failed to parse e2ee config as peer", err)
-		if len(addArgs.endpoint) > 0 {
-			if !addArgs.outbound {
-				err = clientPeerConfigRelay.SetEndpoint(addArgs.endpoint)
-				check("failed to set endpoint", err)
+		if len(addArgs.outboundEndpoint) == 0 {
```

Comment:
Since `addArgs.endpoint` and `addArgs.outboundEndpoint` are already guaranteed to have exactly one set, check for the presence of the relevant flag instead of the absence of the irrelevant flag. 
```suggestion
		if len(addArgs.endpoint) > 0 {
```